7. Dynamic Programming/cookie_collect/[CPP]cookie_collect_csubhradipta.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+ * @file: [CPP]cookie_collect_csubhradipta.cpp 
+ * @brief: Find maximum numbers of cookies can be collected
+ * @details:
+ * Add a row of zeroes to top and bottom of the matrix, for ease of checking the maximum elements
+ * Then traverse matrix left to right from 2nd to nth column
+ * Find the maximum out of left-up, left & left-down of previous column and sum it to the current element
+ * The maximum element of the last column is the maximum cookies that can be collected
+ * Space Complexity : O(n^2)
+ * Time Complexity : O(n^2)
+ * @author [Subhradipta Choudhury](https://github.com/csubhradipta)
+ */
+
+#include <bits/stdc++.h> 
+using namespace std; 
+
+void solve(){
+    int n;
+    cin>>n;
+    int cookies[n+2][n], maxCookies[n];

```

Comment:
Can you do this without using this extra 1D array?

Reply:
> Can you do this without using this extra 1D array?

Yes Done!
Code Updated.